client/context.go

```diff
@@ -324,9 +331,9 @@ func GetFromFields(kr keyring.Keyring, from string, genOnly bool) (sdk.AccAddres
 }
 
 func newKeyringFromFlags(ctx Context, backend string) (keyring.Keyring, error) {
-	if ctx.GenerateOnly {
-		return keyring.New(sdk.KeyringServiceName(), keyring.BackendMemory, ctx.KeyringDir, ctx.Input)
+	if ctx.GenerateOnly || ctx.Simulate {
+		return keyring.New(sdk.KeyringServiceName(), keyring.BackendMemory, ctx.KeyringDir, ctx.Input, ctx.KeyringOptions...)
```

Comment:
Instead of using a context, can we use it in a straight way? Using a context obfuscates a flow. We should use function parameters wherever possible.
The `ReadPersistentCommandFlags` has access to `flagSet` - a set of defined flags. `newKeyringFromFlags` is called in that function. So we can use `flagSet` to carry the information (from config or cli flag) about supported algos. We could also add one more structural parameter to the `ReadPersistentCommandFlags`.

Reply:
Robert, I agree with you on "using a context obfuscates a flow". Fully. I am a big fan of function parameters. I hate any stateful context and I hate even more using contexts to conveniently pack variables in order to reduce the number of function's arguments because that means that you are keen to sacrifice transparency and ease of debugging for the lazy developer's sake.

And yet, this is what we have now. I like your proposed changes, but I believe they are out of scope here. We should open another issue and try to overhaul the Context in a more holistic way.

Reply:
OK, but I would like to avoid adding more `XOptions` directly into config, and I'm worried that someone else from other team will be motivated to do it (by following an example of what's already implemented)